notes/2025-xx-xx-insecurities-remedies-i.md

```diff
@@ -0,0 +1,410 @@
+---
+title: "NixOS Security I: Systemd"
+date: 2025-xx-xx
+description: First installment to a series on securing your NixOS systems
+wordcount: process-anyway
+---
+
+# NixOS Security I: Systemd
+
+A Linux system is a complex ecosystem of components, each with its own set of
+vulnerabilities. NixOS is no exception. While its declarative nature provides
+some advantages, it is not--nor can it ever be--a silver bullet against the
+inherent insecurities that exist in any system. This _status quo_--a state of
+perpetual vulnerability--calls you to action.
+
+For the last 6 months or so, I have been focusing on hardening each and every
+single component of my NixOS installation. This post, as an attempt to document
+my experiences for the sake of establishing a point of reference, marks the
+beginning of a series dedicated to hardening the different components of a NixOS
+system. Given that NixOS is a systemd-based distribution, we’ll start by
+focusing on systemd. Over the course of the series, I’ll also delve into kernel
+and network security, although these topics require further research and are
+beyond the scope of this post.
+
+In this installment, we’ll explore how you can harden various systemd services
+on your system to reduce potential attack surfaces.
+
+## Resources
+
+Before I talk about systemd, I am leaving here the resources I have consulted in
+the past. This post aims to serve as a digestible summary of those resources as
+well as a guide to hardening your system, but do feel free to consult them at
+any time. Most of them extend beyond the scope of Systemd hardening, and will
+come up again in future posts. Visit them at your own discretion, you are sure
+to learn something new.
+
```

Comment:
I'd move the resources to the bottom of the article similar to a bibliography or citations section. Then I'd include a link to that section from here. I think it sticks with the traditional layout for academic articles and increases flow for the reader.

Reply:
Pandoc probably has something to manage bibliographies, but this is not a bibliography in the traditional sense. I'll probably add this as a subsection to the conclusion section.